chronicles.nim

```diff
@@ -71,10 +71,16 @@ macro logScopeIMPL(prevScopes: typed,
     result.add chroniclesExportNode
 
   let activeScope = id("activeChroniclesScope", isPublic)
-  result.add quote do:
-    template `activeScope` {.redefine, used.} =
-      `newRevision`
-      `newAssingments`
+  when NimMajor >= 2:
```

Comment:
```suggestion
  when defined(nimHasTemplateRedefinitionPragma):
```

Reply:
I specifically want this to be easy to just grep for `NimMajor` (or similar common keywords across all of them, not special-purpose ones) to find various of these hacks/kludges/workarounds to remove over time.

Reply:
`nimHas` is what we use in lots of other places because it's more reliable than version checks (also works for devel etc) - also makes it easy to look up the specific PR that introduces it into the language when debugging problems with the new feature.

When the time comes to go out on the crusade to explicitly break the library for users of old nim versions, you'll need to grep `nimHas` regardless (this works for any `nimHas`-enabled feature) to make sure the destruction of this mostly harmless code is complete.

Reply:
In what sense is it more reliable in this particular case? Aside from some of the very slightly pre-2.0 development versions which might have adopted a this feature a bit before they were formally called 2.x for `NimMajor` purposes, but which are definitely not worth supporting, the 1.6 to 2.0 transition is exactly when that started.

Even then, the failure mode in that case here is to just not use the new pragma, which is fine.

`devel` carries a major version of 2 and presumably won't go backwards, so it properly triggers the new pragma.